src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -20,26 +18,35 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     internal abstract class SourceMemberMethodSymbol : LocalFunctionOrSourceMemberMethodSymbol, IAttributeTargetSymbol
     {
         // The flags type is used to compact many different bits of information.
-        protected struct Flags
+        private struct Flags
```

Comment:
made private (including the `flags` field).  This helps control access and ensures everyone is going through the same logic.

Reply:
this helps, for example, to ensure that someone doesn't access a flag directly (like IsNullableAnalysisEnabled), when they should be accessing the helper method for that purpose.